src/main/java/hudson/plugins/logparser/LogParserDisplayConsts.java

```diff
@@ -11,9 +13,13 @@
 
     public LogParserDisplayConsts() {
         // Color of each status
-        colorTable.put(LogParserConsts.ERROR, "red");
-        colorTable.put(LogParserConsts.WARNING, "orange");
-        colorTable.put(LogParserConsts.INFO, "blue");
+        LogParserPublisher.DescriptorImpl descriptor = (LogParserPublisher.DescriptorImpl) Jenkins.getInstance().getDescriptor(LogParserPublisher.class);
+        String colorError = descriptor.getColorError();
+        String colorWarning = descriptor.getColorWarning();
+        String colorInfo = descriptor.getColorInfo();
+        colorTable.put(LogParserConsts.ERROR, colorError == null || colorError.isEmpty() ? "red" : colorError);
```

Comment:
I would think the check for null or isEmpty should be done in the descriptor method so you don't have to check it everywhere else. Check in the getColorError for null or isEmpty and set a sane default ("red' for example). Then you just call the method everywhere and it handles it for you.